runtime-components/non-repudiation-core/src/main/java/com/daml/nonrepudiation/Headers.java

```diff
@@ -10,13 +10,13 @@
     private Headers() {
     }
 
-    public static final Metadata.Key<String> SIGNATURE =
-            Metadata.Key.of("signature", Metadata.ASCII_STRING_MARSHALLER);
+    public static final Metadata.Key<byte[]> SIGNATURE =
```

Comment:
Is there a good reason to have a separate header name for this at this point given that we don’t care about backwards compat?

Reply:
I would have kept it the same, but gRPC actually [throws an exception](https://github.com/grpc/grpc-java/blob/a6df2b2ff403f9aa9097e3fdc6908a53a8f3c4bc/api/src/main/java/io/grpc/Metadata.java#L851-L860) if I don't append `-bin` to the name of binary headers.

Reply:
huh, interesting, makes sense thanks!